src/coreComponents/constitutive/contact/BartonBandis.hpp

```diff
@@ -127,12 +133,13 @@ GEOS_HOST_DEVICE
 GEOS_FORCE_INLINE
 real64 BartonBandisUpdates::computeHydraulicAperture( real64 const aperture,
                                                       real64 const normalTraction,
+                                                      integer const fractureState,
                                                       real64 & dHydraulicAperture_aperture,
                                                       real64 & dHydraulicAperture_dNormalStress ) const
 {
-  real64 const hydraulicAperture = ( aperture >= 0.0 ) ? (aperture + m_aperture0) : m_aperture0 / ( 1 + 9*normalTraction/m_referenceNormalStress );
-  dHydraulicAperture_dNormalStress = ( aperture >= 0.0 ) ? 0.0 : -hydraulicAperture / ( 1 + 9*normalTraction/m_referenceNormalStress ) * 9/m_referenceNormalStress;
-  dHydraulicAperture_aperture = ( aperture >= 0.0 ) ? 1.0 : 0.0;
+  real64 const hydraulicAperture = ( fractureState >= 3.0 ) ? (aperture + m_aperture0) : m_aperture0 / ( 1 + 9*normalTraction/m_referenceNormalStress );
```

Comment:
We may want to be cautious about the convention of the `normalTraction` here. Here `normalTraction` is defined as positive for compression.

Reply:
Yes, we need to ensure that the sign convention for `normalTraction` and `referenceNormalStress` is consistent.

Reply:
`referenceNormalStress` is enforced to be positive and an error check for input is added